crates/ruff_linter/src/rules/flake8_pyi/rules/no_return_argument_annotation.rs

```diff
@@ -9,27 +9,25 @@ use crate::checkers::ast::Checker;
 use crate::settings::types::PythonVersion::Py311;
 
 /// ## What it does
-/// Checks for uses of `typing.NoReturn` (and `typing_extensions.NoReturn`) in
-/// stubs.
+/// Checks for uses of `typing.NoReturn` (and `typing_extensions.NoReturn`) for
+/// parameter annotations.
 ///
 /// ## Why is this bad?
-/// Prefer `typing.Never` (or `typing_extensions.Never`) over `typing.NoReturn`,
-/// as the former is more explicit about the intent of the annotation. This is
-/// a purely stylistic choice, as the two are semantically equivalent.
+/// Prefer `typing.Never` (or `typing_extensions.Never`) over `typing.NoReturn`
+/// for parameter annotations. This is a purely stylistic choice, as the two
+/// are semantically equivalent.
```

Comment:
I think this misses an explanation of why it's more idiomatic (or a preferred stylistic choice) to use  `Never` over `NoReturn`. I assume it's mainly that `NoReturn` is intended for function return types and everything else should use `Never`?

Reply:
It's basically just because it's really confusing to see a type called `NoReturn` in a parameter annotation -- that's the only reason we added the `Never` type in Python 3.11 😄 It has exactly the same semantics as `Noreturn`.

I'll try to flesh this out a little more...